src/org/sosy_lab/java_smt/solvers/bitwuzla/BitwuzlaSolverContext.java

```diff
@@ -0,0 +1,266 @@
+      BitwuzlaFormulaCreator pCreator,
+      ShutdownNotifier pShutdownNotifier,
+      Options pOptions) {
+    super(pManager);
+    manager = pManager;
+    creator = pCreator;
+    shutdownNotifier = pShutdownNotifier;
+    solverOptions = pOptions;
+  }
+
+  @SuppressWarnings("unused")
+  public static BitwuzlaSolverContext create(
+      Configuration config,
+      ShutdownNotifier pShutdownNotifier,
+      @Nullable PathCounterTemplate solverLogfile,
+      long randomSeed,
+      FloatingPointRoundingMode pFloatingPointRoundingMode,
+      Consumer<String> pLoader)
+      throws InvalidConfigurationException {
+    pLoader.accept("bitwuzlaJNI");
+
+    TermManager termManager = new TermManager();
+    Options solverOptions = buildBitwuzlaOptions(new BitwuzlaSettings(config), randomSeed);
+
+    BitwuzlaFormulaCreator creator = new BitwuzlaFormulaCreator(termManager);
+    BitwuzlaUFManager functionTheory = new BitwuzlaUFManager(creator);
+    BitwuzlaBooleanFormulaManager booleanTheory = new BitwuzlaBooleanFormulaManager(creator);
+    BitwuzlaBitvectorFormulaManager bitvectorTheory =
+        new BitwuzlaBitvectorFormulaManager(creator, booleanTheory);
+    BitwuzlaQuantifiedFormulaManager quantifierTheory =
+        new BitwuzlaQuantifiedFormulaManager(creator);
+    BitwuzlaFloatingPointManager floatingPointTheory =
+        new BitwuzlaFloatingPointManager(creator, pFloatingPointRoundingMode);
+    BitwuzlaArrayFormulaManager arrayTheory = new BitwuzlaArrayFormulaManager(creator);
+    BitwuzlaFormulaManager manager =
+        new BitwuzlaFormulaManager(
+            creator,
+            functionTheory,
+            booleanTheory,
+            bitvectorTheory,
+            quantifierTheory,
+            floatingPointTheory,
+            arrayTheory,
+            solverOptions);
+
+    return new BitwuzlaSolverContext(manager, creator, pShutdownNotifier, solverOptions);
+  }
+
+  /**
+   * Set more options for Bitwuzla.
+   *
+   * @param pOptions current Bitwuzla options
+   * @param pFurtherOptions String to be parsed with options to be set.
+   * @throws InvalidConfigurationException signals that the format of the option string is wrong or
+   *     an invalid option is used.
+   */
+  private static Options setFurtherOptions(Options pOptions, String pFurtherOptions)
+      throws InvalidConfigurationException {
+    MapSplitter optionSplitter =
+        Splitter.on(',')
+            .trimResults()
+            .omitEmptyStrings()
+            .withKeyValueSeparator(Splitter.on('=').limit(2).trimResults());
+    ImmutableMap<String, String> furtherOptionsMap;
+
+    try {
+      furtherOptionsMap = ImmutableMap.copyOf(optionSplitter.split(pFurtherOptions));
+    } catch (IllegalArgumentException e) {
+      throw new InvalidConfigurationException(
+          "Invalid Bitwuzla option in \"" + pFurtherOptions + "\": " + e.getMessage(), e);
+    }
+    for (Map.Entry<String, String> option : furtherOptionsMap.entrySet()) {
```

Comment:
Reflection-based setting of options is no clean solution.
The alternative would be a larger switch-case that sets each option by name.
Hmmm... It might be working, so we could keep it.

Reply:
Aren't we setting all configs via reflections...? :D

Reply:
Fixed.